File: Math/Greatest Common Divisor/gcd.java
import java.util.Scanner;

public class Main {
  
  public static int gcd(int a, int b) { // Change type if needed
    if(b == 0)
      return a;
    
    return gcd(b, a % b);
  }
    
  public static void main(String[] args) {
    Scanner sc = new Scanner(System.in);
    int a, b;
    
    a = sc.nextInt();
    b = sc.nextInt();
    System.out.println(gcd(a, b));
    
  }
}

File: Algorithms/Math/Number Theory/Greatest Common Divisor/gcd.cpp
#include <iostream>

using namespace std;

typedef int T; // Change type if needed

T gcd(T a, T b) {
  if(b == 0)
    return a;
  
  return gcd(b, a % b);
}

int main() {
  T a, b;
  
  cin >> a >> b;
  cout << gcd(a, b) << endl;
  
  return 0;
}
